src/client/pydaos/raw/daos_api.py

```diff
@@ -1663,19 +1662,37 @@ def close(self, coh=None, cb_func=None):
                                             self))
             thread.start()
 
-    def query(self, coh=None, cb_func=None):
-        """Query container information."""
+    def query(self, coh=None, daos_property=None, cb_func=None):
+        """Query container information.
+
+        Args:
+            coh (ctypes.c_uint64, optional): Container handle. Defaults to None.
+            daos_property (DaosContProperties, optional): Empty data structure that
+                stores the property values. The values are set during create(). The
+                configuration of the data structure depends on the property values set.
+                See create() for more details. Defaults to None.
+            cb_func (CallbackHandler, None): Callback function. Defaults to None.
+
+        Returns:
+            daos_cref.ContInfo: Container info that stores UUID, snapshots, redundancy
+                factor, etc.
+
+        """
```

Comment:
A few comments:
1. It seems this is actually a `DaosProperty`, not `DaosContProperties`, right?
2. The C API is
    ```
    int
    daos_cont_query(daos_handle_t coh, daos_cont_info_t *info,
                    daos_prop_t *cont_prop, daos_event_t *ev);
    ```
    If it's not too much trouble, could we rename `daos_property` more similar to the C API? I.e. `cont_prop`
3. I'm not familiar with the pydaos API, but do we need to support backwards compatibility with the function definitions? Inserting a new param before `cb_func` isn't backwards compatible if someone uses positional arguments.

Reply:
Thanks for the comment. I updated the variable name and moved the properties to the yaml.
For 3, I searched and noticed that no one uses positional arguments, and I ran those tests that calls `query()`, so I believe it's fine.

Reply:
We expose the `pydaos` API to users, but I'm not sure if the `pydaos.raw` API is expected to be backwards compatible:
https://docs.daos.io/v2.0/user/python/#Native_Programming_Interface

Reply:
It might be okay, since I think the `pydaos.raw` API is primarily for our internal testing

Reply:
That's a good point, but I noticed that `cb_func` comes last in all other methods, so I think it's better to be consistent. In addition, I think we should refactor this API if we expect external users to use it.

Reply:
Fair enough. I agree it needs work before external users use it.